### PR TITLE
Add string padding functions (pad_left and pad_right)

### PR DESCRIPTION
**Pull request by mavam**
## Summary
- Adds two new string functions: `pad_left` and `pad_right` for padding strings to a specified length
- Both functions support custom padding characters (default: space)
- Proper UTF-8 character counting support

## Changes
- Added `pad` class in `libtenzir/builtins/functions/string.cpp` implementing both padding functions
- Created documentation for both functions in `docs/functions/`
- Added comprehensive integration tests with edge cases

## Usage Examples
```tql
from {x: "hello".pad_left(10)}           // "     hello"
from {x: "42".pad_left(5, "0")}          // "00042"
from {x: "hello".pad_right(10, ".")}     // "hello....."
from {x: "test".pad_left(8, "★")}        // "★★★★test"
```

## Test Plan
- [x] Integration tests added for both functions
- [x] Tests cover edge cases: empty strings, zero/negative lengths, multi-character padding (error), null handling
- [x] UTF-8 character support tested with unicode padding characters
- [x] All tests passing

🤖 Generated with [Claude Code](https://claude.ai/code)

### CODE REVIEW

File: libtenzir/builtins/functions/string.cpp [hunk 1]
```diff
@@ -211,6 +211,136 @@ class trim : public virtual function_plugin {
...
+              }
+              check(b.Append(result));
+            }
+          },
+          [&](const auto&, const auto&) {
+            // Type mismatch
+            diagnostic::warning("`{}` expected (string, int), but got ({}, {})",
+                                name, subject.type.kind(), length.type.kind())
+              .primary(subject_expr)
+              .emit(ctx);
```
Comment (jachris) [inline libtenzir/builtins/functions/string.cpp:328]: We normally report type errors for a single parameter instead of combining everything. Then you can also use the correct `primary` (it's not correct right now if `length` has the bad type).

File: tenzir/tests/exec/functions/string/pad_left.txt [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+  multi_pad: null,
+  null_str: null,
+  null_len: null,
+}
+warning: `pad_left` expected single character for padding, but got `ab` with length 2
+  --> exec/functions/string/pad_left.tql:19:14
+   |
+19 |   multi_pad: "test".pad_left(10, "ab"),
+   |              ~~~~~~ 
+   |
```
Comment (jachris) [inline tenzir/tests/exec/functions/string/pad_left.txt:19]: Diagnostic point to "ab".

File: tenzir/tests/exec/functions/string/pad_left.txt [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+warning: `pad_left` expected (string, int), but got (null, int64)
+  --> exec/functions/string/pad_left.tql:21:13
+   |
+21 |   null_str: null.pad_left(5),
+   |             ~~~~ 
+   |
+
```
Comment (jachris) [inline tenzir/tests/exec/functions/string/pad_left.txt:27]: Your function doesn't handle `null_type` correctly. `null` should have the same behavior as `null.string()`.

File: tenzir/tests/exec/functions/string/pad_left.txt (hunk 2 above)
Comment (mavam) [reply tenzir/tests/exec/functions/string/pad_left.txt]: Fixed in 4928fac248c

File: tenzir/tests/exec/functions/string/pad_left.txt (hunk 3 above)
Comment (mavam) [reply tenzir/tests/exec/functions/string/pad_left.txt]: Changed in 767ad4a29e7.

File: changelog/changes/crash-prime-aboard.md [hunk 4]
```diff
@@ -0,0 +1,81 @@
+---
+title: "New string padding functions"
```
Comment (dominiklohmann) [inline changelog/changes/crash-prime-aboard.md:2]: We used `trim_start` and `trim_end`, as it is agnostic to the direction of writing. Should these be called `pad_start` and `pad_end` instead?

Comment (mavam) [reply changelog/changes/crash-prime-aboard.md:2]: Yes, good point. That said, I find "left" and "right" a lot more intuitive. Any objections to renaming the trim functions to `trim_left` and `trim_right`?

File: libtenzir/builtins/functions/string.cpp [hunk 5]
```diff
@@ -211,6 +211,147 @@ class trim : public virtual function_plugin {
...
+                  check(b.AppendNull());
+                  continue;
+                }
+                auto str = subject_array.GetView(i);
+                auto target_length
+                  = detail::narrow<int64_t>(length_array.Value(i));
+                if (target_length <= 0) {
+                  check(b.Append(""));
+                  continue;
+                }
```
[the remark this answers is not in the document]
Comment (mavam) [reply libtenzir/builtins/functions/string.cpp]: Great catch. Our tests previously were wrong. Removing the check exposed the flow:
```
from {
  // Basic padding with default space
  basic: "hello".pad_left(10),
  // Padding with custom character
  zeros: "42".pad_left(5, "0"),
  // String already long enough
  no_pad: "hello world".pad_left(5),
  // Empty string
  empty: "".pad_left(5, "-"),
  // Single character
  single: "x".pad_left(3, "."),
  // Padding with unicode character
  unicode: "test".pad_left(8, "★"),
  // Zero length
  zero_len: "hello".pad_left(0),
  // Negative length (should return empty)
  neg_len: "hello".pad_left(-5),
  // Null handling
  null_str: null.pad_left(5),
  null_len: "test".pad_left(null),
}
```

```
✘ exec/functions/string/pad_left.tql
┌─▶ exec/functions/string/pad_left.txt
│    single: "..x",
│    unicode: "★★★★test",
│ -  zero_len: "",
│ -  neg_len: "",
│ +  zero_len: "hello",
│ +  neg_len: "hello",
│    null_str: null,
└─   null_len: null,
i 2/4 tests passed (0 skipped)
```

The new behavior is what we want. I updated the baseline.

File: changelog/changes/crash-prime-aboard.md (hunk 4 above)
Comment (mavam) [reply changelog/changes/crash-prime-aboard.md:2]: Okay, given that other languages do this as well, I'll go with `pad_start` and `pad_end`.
